packages/mock-block-dock/dev/test-block.tsx

```diff
@@ -1,14 +1,44 @@
-import { BlockComponent } from "blockprotocol/react";
+import { BlockComponent, useGraphBlockService } from "@blockprotocol/graph";
 import * as React from "react";
+import { useRef } from "react";
 
 type AppProps = {
   name: string;
 };
 
-export const TestBlock: BlockComponent<AppProps> = ({ entityId, name }) => {
+export const TestBlock: BlockComponent<AppProps> = ({ graph }) => {
+  const {
+    blockEntity: { entityId, properties },
+  } = graph;
+  const blockRef = useRef<HTMLDivElement>(null);
+
+  const { graphService } = useGraphBlockService({ ref: blockRef });
+
   return (
-    <div>
-      Hello {name}! The id of this block is {entityId}
+    <div ref={blockRef}>
+      <h1>
+        Hello {properties.name}! The id of this block is {entityId}
+      </h1>
+      <input
+        type="text"
+        placeholder="This block's entity's 'name' property"
+        value={properties.name}
+        onChange={async (event) => {
+          try {
+            const { data, errors } = await graphService!.updateEntity({
```

Comment:
Unsure if this is supposed to serve as a good example to other users, or just as a little testing tool, but you've got the potential for race conditions because this is handle async - you probably want to set up a queue or something. That reminds me, is the value in `properties` gauranteed to be updated by the time `updateEntity` is finished. Like what's the order in terms of re-rendering with new `properties` vs the promise from `updateEntity` resolving?